### PR TITLE
Math Magicians: tests

### PR DESCRIPTION
**Pull request by mustabbas**
### Change Done 
- [ ] Set up React Testing Library.
- [ ] Added unit tests for the files operate.js and calculate.js using Jest.
- [ ] Created unit tests for all React components:
- [ ] Run full tests suite using Jest. All tests pass.

### CODE REVIEW

File: src/logic/logic.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import operate from './operate.js';
+import calculate from './calculate.js';
+
+describe('test for logic of operate', () => {
+  test('Display the Addition of 2 number ', () => {
+    expect(operate(2, 4, '+')).toBe('6');
+  });
```
Comment (GzimAsani) [inline src/logic/logic.test.js]: - [ ] There should be the usage of Jest snapshots for testing all the components.

Comment (GzimAsani) [inline src/logic/logic.test.js]: - [ ] All the functionalities & scenarios of the `operate.js` and `calculate.js` modules should be covered using Jest. Currently, you covered only some of them.

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi @mustabbas and @Meltrust                                     
Good job so far! Your code looks fine, good job :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Gitflow is used :clap:
- Good PR description :clap: 

## Required Changes ♻️ 
- [ ] _Check inline comments for more information :arrow_double_down:_ 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/logic.test.js (hunk 1 above)
Comment (mustabbas) [reply src/logic/logic.test.js]: Hi, thanks for this feedback but we are doing Jest snapshots for all the components in  [component.test.js ](https://github.com/mustabbas/calculator/blob/test/src/component/component.test.js).
https://github.com/mustabbas/calculator/blob/5d35b80b48ac4a53a866b218818b7b672a19e5ae/src/component/component.test.js#L9-L40
please check that to be sure I am understand you correct Thanks

Comment (mustabbas) [reply src/logic/logic.test.js]: Hi, thanks for this feedback, we will do the remaining Thanks

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉🎉

## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/logic.test.js (hunk 1 above)
Comment (DeliceLydia) [reply src/logic/logic.test.js]: Addressed this

**Pull request merged**